kvmagent/kvmagent/plugins/vm_plugin.py

```
@@ -6731,6 +6735,26 @@ def make_mdev_device(mdevUuids):
                 # convert mdevUuid to 8-4-4-4-12 format
                 e(source, "address", None, { "uuid": uuidhelper.to_full_uuid(mdevUuid) })
 
+        def make_shmem_device(shmemDevices):
+            root = elements['root']
+            qcmd = root.find('{http://libvirt.org/schemas/domain/qemu/1.0}commandline')
+            if qcmd is None:
+                qcmd = e(root, 'qemu:commandline')
+            for idx, shmem in enumerate(shmemDevices):
+                mem_path = shmem.path
+                if not mem_path:
+                    raise kvmagent.KvmError('dGPU shmem path is required but missing in StartVmCmd.addons[pciDevice.dgpu]')
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)